### PR TITLE
Breaking: Migrate to material-ui v4

### LINKED ISSUE #67: Migrate to material-ui v4

### PR DESCRIPTION
**Pull request by Rolikasi**
Closes #67

### CODE REVIEW

**Comment by Rolikasi** [comment]
@helfi92, should I change test snapshot files?

File: package.json [hunk 1]
```diff
@@ -53,9 +53,9 @@
     "webpack-dev-server": "^3.2.1"
   },
   "peerDependencies": {
-    "@material-ui/core": "^3.0.0",
-    "react": "^16.4.1",
-    "react-dom": "^16.4.1"
+    "@material-ui/core": "^4.0.0",
+    "react": "^16.10.2",
+    "react-dom": "^16.10.2"
```
Comment (helfi92) [inline package.json]: Let's make the peer dependencies for react and react-dom to be `^16.8.0`, just enough to support react hooks.

File: src/components/Code/index.jsx [hunk 2]
```diff
@@ -11,6 +12,11 @@ const validLanguage = (props, propName) => {
     return new Error(`Language '${language}' not supported by highlight.js`);
   }
 };
+@withStyles({
+  textCode: {
+    color: 'black',
```
Comment (helfi92) [inline src/components/Code/index.jsx]: Do you know why this is needed now?

File: src/components/ErrorPanel/index.jsx [hunk 3]
```diff
@@ -164,7 +168,11 @@ export default class ErrorPanel extends Component {
         )}
         disabled={!showStack}>
         <ExpansionPanelSummary
-          classes={{ disabled: classes.disabled }}
+          classes={{
+            disabled: classes.disabled,
+            expandIcon: classes.expandIcon,
+          }}
+          IconButtonProps={{ edge: false }}
```
Comment (helfi92) [inline src/components/ErrorPanel/index.jsx]: `IconButtonProps={{ edge: 'end' }}` then you can remove `classes.expandIcon`.

File: src/theme.js [hunk 4]
```diff
@@ -18,20 +18,26 @@ const theme = createMuiTheme({
...
     caption: Roboto400,
     button: Roboto500,
   },
+  overrides: {
+    MuiButton: {
+      root: {
+        lineHeight: 1.5,
+      },
+    },
+  },
```
Comment (helfi92) [inline src/theme.js]: We can remove the MuiButton override. I know they changed it to 1.75 but that's okay.

File: src/components/ErrorPanel/index.jsx (hunk 3 above)
Comment (Rolikasi) [reply src/components/ErrorPanel/index.jsx]: classes.expandIcon from @withStyles, after migration arrow icon is to far from edge, I added css rules that work like in previous version
right: 8px,
position: absolute

File: src/components/Code/index.jsx (hunk 2 above)
Comment (Rolikasi) [reply src/components/Code/index.jsx]: After migration we've got new css rule in body

css rule

Color is inherited from it.
I think this solution is not good, if `code` will be added to "dark" site, color will stay black I guess.
What can we do here?

File: src/components/ErrorPanel/index.jsx (hunk 3 above)
Comment (Rolikasi) [reply src/components/ErrorPanel/index.jsx]: Ah!!! I didn't get it! Lol

Comment (Rolikasi) [reply src/components/ErrorPanel/index.jsx]: Also it's edge: "end" by default) 
I will remove IconButtonProps
The difference is 4px between v3 and v4. Is it ok?

Comment (helfi92) [reply src/components/ErrorPanel/index.jsx]: Yeah, I think it's ok.

File: src/components/Code/index.jsx (hunk 2 above)
Comment (Rolikasi) [reply src/components/Code/index.jsx]: Perfect! thanks! 👍

File: package.json [hunk 5]
```diff
@@ -60,6 +58,8 @@
   "dependencies": {
     "classnames": "^2.2.6",
     "codemirror": "^5.39.0",
+    "enzyme": "^3.10.0",
+    "enzyme-adapter-react-16": "^1.15.1",
```
Comment (helfi92) [inline package.json]: These should be in `devDependencies`. They are only used for tests.

File: package.json [hunk 6]
```diff
@@ -74,6 +74,7 @@
     "react-codemirror2": "^5.1.0",
     "react-fout-stager": "^3.0.0",
     "react-router-dom": "^4.3.1",
+    "react-test-renderer": "^16.11.0",
```
Comment (helfi92) [inline package.json]: Does it work without this? We didn't have this library before.

File: package.json (hunk 5 above)
Comment (Rolikasi) [reply package.json]: I realized it right after git push. Just a sec

File: package.json (hunk 6 above)
Comment (Rolikasi) [reply package.json]: Yes, we can safely delete this

**Pull request merged**